### ISSUE TITLE
F.cross_entropy and nn.CrossEntropy raise exception if no batch dimension is provided

### ISSUE THREAD

**Issue by abaisero** [pytorch/pytorch#76939]
### 🐛 Describe the bug

pytorch version 1.11 should have switched to allowing 0-batch dimensions in many of its functions and losses, including the cross-entropy ones.  This is reflected in the documentation and in the relevant PR #71055, however, actually using them without batch dimensions will result in an IndexError being raised.

```
import torch
import torch.nn.functional as F

# works
shape = (3, 5)
input = torch.randn(*shape).log_softmax(dim=-1)
target = torch.randn(*shape).softmax(dim=-1)
F.cross_entropy(input, target)

# crashes
shape = (3,)
input = torch.randn(*shape).log_softmax(dim=-1)
target = torch.randn(*shape).softmax(dim=-1)
F.cross_entropy(input, target)
# IndexError:  Dimension out of range(expected to be in range of [-1, 0], but got 1)
```

### Versions

Collecting environment information...
PyTorch version: 1.11.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 10 (buster) (x86_64)
GCC version: (Debian 8.3.0-6) 8.3.0
Clang version: 7.0.1-8+deb10u2 (tags/RELEASE_701/final)
CMake version: version 3.13.4
Libc version: glibc-2.10

Python version: 3.7.9 (default, Aug 31 2020, 12:42:55)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-4.19.0-20-amd64-x86_64-with-debian-10.12
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.931
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.0
[pip3] numpydoc==1.1.0
[pip3] torch==1.11.0
[pip3] torchvision==0.4.2
[conda] _pytorch_select           0.2                       gpu_0  
[conda] blas                      1.0                         mkl  
[conda] cpuonly                   1.0                           0    pytorch
[conda] cudatoolkit               10.0.130                      0  
[conda] mkl                       2020.2                      256  
[conda] mkl-service               2.3.0            py37he8ac12f_0  
[conda] mkl_fft                   1.2.0            py37h23d657b_0  
[conda] mkl_random                1.1.1            py37h0573a6f_0  
[conda] mypy                      0.931                    pypi_0    pypi
[conda] mypy_extensions           0.4.3                    py37_0  
[conda] numpy                     1.21.0                   pypi_0    pypi
[conda] numpy-base                1.19.2           py37hfa32c7d_0  
[conda] numpydoc                  1.1.0              pyhd3eb1b0_1  
[conda] torch                     1.11.0                   pypi_0    pypi
[conda] torchvision               0.4.2           cuda100py37hecfc37a_0

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by jbschlosser** [comment]
Hey @abaisero, thanks for the report! Looks like no-batch-dim inputs are not supported for the case of class probabilities as targets. We'd accept a PR fixing this omission.
https://github.com/pytorch/pytorch/blob/4ded63e2acaf0d04defcd32d40e3d07f753e37c7/aten/src/ATen/native/LossNLL.cpp#L460-L522

**Comment by jbschlosser** [comment]
Fixed in #77653.
